### DISCUSSION TITLE
Why is the exact data type not passed to the onChange handler?

### DISCUSSION THREAD

**Discussion by 1natsu172** [mantinedev/mantine#6401] [Feedback]
Mantine has helped me reduce the cost of UI development, thank you. I am writing to provide feedback and ask a question about parts where the compatibility with TypeScript seems low.

As the title, why is the value type passed as an argument to the onChange handler fixed as a `string`? This problem is seen with like `SegmentedControls`, `MultiSelect`, etc. I'm implementing the my-handlers with exact types, so having to cast the value in every implementation is very cumbersome.

```tsx
// typeof value => "left" | "right" | "center"
<SegmentedControl
  data={[{label: "LEFT", value: "left"}, {label: "RIGHT", value: "right"}, {label: "CENTER", value: "center"}]}
  value={value}
  // `e` is `string`
  onChange={(e) => {
    myHandler(e as typeof value);
  }}
/>;
```

The onChange handler should be able to infer the type based on the `data` props, so I am genuinely curious why it is not implemented that way.

(This is not an urgent discussion 👐  )

**Comment by rtivital** [comment]
I do not see a reason to add this feature. It is very rarely used but it requires significant efforts to implement and maintain.

**Comment by 1natsu172** [comment]
I understand the importance of maintaining the code, so if this is the decided approach, that's fine. Thanks for your insight.
